### PR TITLE
fix generate api dont take 2d attention mask bug

### PR DESCRIPTION
**Pull request by smallv0221**
### PR types

Bug fixes
### PR changes

APIs
### Description

fix generate api dont take 2d attention mask bug

### CODE REVIEW

File: paddlenlp/transformers/gpt/modeling.py [hunk 1]
```diff
@@ -811,10 +811,11 @@ def forward(self,
             diagonal=1)
 
         if attention_mask is not None:
+            attention_mask = attention_mask[:, None, None, :]
             attention_mask = attention_mask + causal_mask
         else:
             attention_mask = causal_mask
-
+        print(attention_mask)
```
Comment (ZeyuChen) [inline paddlenlp/transformers/gpt/modeling.py]: 这个print不太好吧？

File: paddlenlp/transformers/generation_utils.py [hunk 2]
```diff
@@ -422,17 +422,27 @@ def update_model_kwargs_for_generation(outputs,
...
-            dtype = convert_dtype(attention_mask.dtype)
-            if 'int' in dtype:
-                attention_mask[:, :, -1, -1] = 1
-            elif 'float' in dtype:
-                attention_mask[:, :, -1, -1] = 0.0
+            if len(attention_mask.shape) == 4:
+                attention_mask = nn.Pad2D(
+                    [0, 0, 0, 1], mode='replicate')(attention_mask)
+                attention_mask = nn.Pad2D(
+                    [0, 1, 0, 0], value=-1e9)(attention_mask)
```
Comment (gongel) [inline paddlenlp/transformers/generation_utils.py]: 这里需要统一改成`-1e4`吗？参考PR：#1585

File: paddlenlp/transformers/gpt/modeling.py [hunk 3]
```diff
@@ -1151,13 +1152,16 @@ def prepare_inputs_for_generation(self,
         # only last token for inputs_ids if cache is defined in kwargs
         position_ids = kwargs.get("position_ids", None)
         attention_mask = kwargs.get("attention_mask", None)
+        if attention_mask is not None:
+            if len(attention_mask.shape) == 4:
+                attention_mask = attention_mask[:, -1, -1, :]
+            if "int" in paddle.fluid.data_feeder.convert_dtype(
+                    attention_mask.dtype):
+                attention_mask = (1.0 - attention_mask) * -1e9
```
Comment (gongel) [inline paddlenlp/transformers/gpt/modeling.py]: 同上

File: paddlenlp/transformers/generation_utils.py (hunk 2 above)
Comment (smallv0221) [reply paddlenlp/transformers/generation_utils.py]: done！

File: paddlenlp/transformers/gpt/modeling.py (hunk 1 above)
Comment (smallv0221) [reply paddlenlp/transformers/gpt/modeling.py]: done！

**Review by gongel — APPROVED** [review]
